product_supplierinfo_group/models/product_supplierinfo.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2020 Akretion France (http://www.akretion.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from odoo import api, fields, models
+from copy import deepcopy
+
+# format: field_from_supplierinfo:field_from_group
+MAPPING_RELATED = {
+    "product_tmpl_id": "product_tmpl_id",
+    "name": "partner_id",
+    "product_id": "product_id",
+    "product_name": "product_name",
+    "product_code": "product_code",
+    "sequence": "sequence",
+}
+
+MAPPING_MATCH_GROUP = {
+    "product_tmpl_id": "product_tmpl_id",
+    "name": "partner_id",
+    "product_id": "product_id",
+    "product_name": "product_name",
+    "product_code": "product_code",
+}
+_logger = logging.getLogger(__name__)
+
+
+class ProductSupplierinfo(models.Model):
+    _inherit = "product.supplierinfo"
+
+    supplierinfo_group_id = fields.Many2one("product.supplierinfo.group", required=True)
+    product_tmpl_id = fields.Many2one(
+        related="supplierinfo_group_id.product_tmpl_id", store=True
+    )
+    name = fields.Many2one(
+        related="supplierinfo_group_id.partner_id", required=False, store=True
```

Comment:
@kevinkhao Here no need to add `required=False`, Because by default `required False`.

Reply:
@nikul-serpentcs thanks for the catch

Reply:
actually we do need to set it to required=False https://github.com/odoo/odoo/blob/d87111ae3e3cf3d654180c986745623ebe731667/addons/product/models/product.py#L743 if we don't then we have problems when trying to create a new record